app/controllers/concerns/active_storage/set_disk_blob.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module ActiveStorage
+  module SetDiskBlob
+    extend ActiveSupport::Concern
+
+    included do
+      before_action :verify_request
+      before_action :set_disk_blob
+    end
+
+    private
+
+    def set_disk_blob
+      return unless @blob.nil?
+
+      @blob = ActiveStorage::Blob.find_by(key: decrypted_hash[:key])
+    rescue StandardError, ActiveSupport::MessageVerifier::InvalidSignature
```

Comment:
You can use `||=` to avoid the nil check, e.g.

``` ruby
@blob ||= begin
  ActiveStorage::Blob.find_by(key: decrypted_hash[:key])
rescue StandardError
  head :not_found
end
```

The exception `ActiveSupport::MessageVerifier::InvalidSignature` is a [descendant][1] of `StandardError` and will match the rescue anyway.

[1]: https://github.com/rails/rails/blob/b9ca94caea2ca6a6cc09abaffaad67b447134079/activesupport/lib/active_support/message_verifier.rb#L104